viz_scripts/scaffolding.py

```diff
@@ -161,6 +161,40 @@ def load_viz_notebook_data(year, month, program, study_type, dic_re, dic_pur=Non
 
     return expanded_ct, file_suffix, quality_text, debug_df
 
+def load_viz_notebook_sensor_inference_data(year, month, program, include_test_users=False, sensed_algo_prefix="cleaned"):
+    """ Inputs:
+    year/month/program = parameters from the visualization notebook
+
+    Pipeline to load and process the data before use in sensor-based visualization notebooks.
+    """
+    tq = get_time_query(year, month)
+    participant_ct_df = load_all_participant_trips(program, tq, include_test_users)
+    expanded_ct = participant_ct_df
+    expanded_ct["primary_mode_non_other"] = participant_ct_df.cleaned_section_summary.apply(lambda md: max(md["distance"], key=md["distance"].get))
+    expanded_ct.primary_mode_non_other.replace({"ON_FOOT": "WALKING"}, inplace=True)
+    valid_sensed_modes = ["WALKING", "BICYCLING", "IN_VEHICLE", "AIR_OR_HSR", "UNKNOWN"]
```

Comment:
Who provides us this list, and how often we might have to change it? Would it be appropriate to store it in some config files and access it from there?
The list name suggests valid_sensed_modes, but we do have an element as "UNKNOWN".

Reply:
We provide ourselves with this list. This is the list of supported modes for the `cleaned_sections`, as outlined in Chapter 5 of my thesis. Changing this list will require changing the implementation of our inference algorithm, which will take multiple months to finalize. Updating this list is the smallest part of that rework.